chainer/optimizers/eve.py

```diff
@@ -0,0 +1,88 @@
+import math
+
+import numpy
+
+from chainer import cuda
+from chainer import optimizer
+
+
+class Eve(optimizer.GradientMethod):
+
+    """Eve optimization algorithm.
+
+    See: https://arxiv.org/abs/1611.01505
+
+    You must speficify ``lossfun`` when you call its ``update`` method so that
+    it can utilize loss values in optimization.
+    """
+
+    def __init__(self, alpha=0.001, beta1=0.9, beta2=0.999, beta3=0.999,
+                 eps=1e-8, k=0.1, K=10):
```

Comment:
I think `lower_threshold` is a good name. If you would like to make the code consistent with the paper, you can use `upper_k` and `lower_k`.